src/components/forms/FieldsCreator.tsx

```diff
@@ -0,0 +1,135 @@
+
+  const onChangeHandle = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    setFieldValue(e.target.name, e.target.value)
+
+    clearTimeout(timer)
+    if (e.target.value) {
+      const changeTimer = setTimeout(() => {
+        handleChangeInput(e)
+      }, 1000)
+      setTimer(changeTimer)
+    }
+  }
+
+  const handleChangeOption = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setFieldValue(e.target.name, e.target.value)
+    handleChange(e)
+  }
+
+  useEffect(() => {
+    handleValueChange(values)
+  }, [values])
+
+  const renderControls = (variable: IFormVariable) => {
+    switch (variable.type) {
+      case "string":
+        return (
+          <StringField variable={variable} onChangeHandle={onChangeHandle} />
+        )
+      // case "number":
+      //   return <NumberField variable={variable} />
+      case "bool":
+        return <BooleanField variable={variable} />
+      case "select":
+        return (
+          <SelectField variable={variable} handleChange={handleChangeOption} />
+        )
+      case "selectradio":
+        return (
+          <SelectRadioField
+            variable={variable}
+            handleChange={handleChangeOption}
+          />
+        )
+      case "dob":
+        return (
+          <DateField
+            variable={variable}
+            handleTargetValueChange={handleTargetValueChange}
+          />
+        )
+      // case "list(string)":
+      //   return <ListField variable={variable} />
+      // case "file":
+      //   return <FilesField variable={variable} />
+      default:
+        return (
+          <StringField variable={variable} onChangeHandle={onChangeHandle} />
+        )
+    }
+  }
+
+  return (
+    <div className="flex-wrap">
+      {Object.keys(groupSortedList).map((groupSortedListKey, indexItem) => {
+        return (
+          <div
+            key={indexItem}
+            className="w-full border border-primary rounded-lg my-2 p-4 transition ease-in-out delay-150"
+          >
+            <label className="text-sm flex gap-1 mb-1">
+              <div className="rounded-full bg-primary w-5 h-5 text-center text-base-100">
+                {indexItem + 1}
+              </div>
+              {groupSortedList[groupSortedListKey][0].question}
+            </label>
+            {groupSortedList[groupSortedListKey].length > 1 ? (
+              <div className="w-full sm:flex gap-2">
+                {groupSortedList[groupSortedListKey].map((variable) => (
+                  <div className="relative w-full" key={variable.questionId}>
+                    {renderControls(variable)}
+                  </div>
+                ))}
+              </div>
+            ) : (
+              groupSortedList[groupSortedListKey].map((variable) => (
+                <div className="relative w-full" key={variable.questionId}>
+                  {renderControls(variable)}
+                </div>
+              ))
+            )}
```

Comment:
Why the length check? Can't this be combined?

Reply:
here we are checking if length more than one then we are combining form fields

Reply:
Wouldn't this code work for **any size**? (0, 1, 2, 3+)

```ts
              <div className="w-full sm:flex gap-2">
                {groupSortedList[groupSortedListKey].map((variable) => (
                  <div className="relative w-full" key={variable.questionId}>
                    {renderControls(variable)}
                  </div>
                ))}
              </div>
              ```
              
              I don't think you need the 2nd block or the length check.